### PR TITLE
sway: 1.8.1 -> 1.9.0

### PR DESCRIPTION
**Pull request by raboof**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by oxalica** [comment]
:tada: It's released now. https://github.com/swaywm/sway/releases/tag/1.9

File: pkgs/by-name/sw/sway-unwrapped/package.nix [hunk 1]
```diff
@@ -11,19 +11,16 @@
...
-  version = "1.8.1";
+  version = "1.9-rc3";
 
   inherit enableXWayland isNixOS systemdSupport trayEnabled;
   src = fetchFromGitHub {
     owner = "swaywm";
     repo = "sway";
     rev = finalAttrs.version;
-    hash = "sha256-WxnT+le9vneQLFPz2KoBduOI+zfZPhn1fKlaqbPL6/g=";
+    hash = "sha256-1NhXFRVV+jMcjs+p8MK+4imt8yBBAEuzl/1aAcS+bPM=";
```
Comment (norpol) [inline pkgs/by-name/sw/sway-unwrapped/package.nix:23]:
```suggestion
    hash = "sha256-/6+iDkQfdLcL/pTJaqNc6QdP4SRVOYLjfOItEu/bZtg=";
```

File: pkgs/by-name/sw/sway-unwrapped/package.nix [hunk 2]
```diff
@@ -11,19 +11,16 @@
 , trayEnabled ? systemdSupport
 }:
 
-let
-  wlroots = wlroots_0_16;
-in
 stdenv.mkDerivation (finalAttrs: {
   pname = "sway-unwrapped";
-  version = "1.8.1";
+  version = "1.9-rc3";
```
Comment (norpol) [inline pkgs/by-name/sw/sway-unwrapped/package.nix:16]:
```suggestion
  version = "1.9";
```

File: pkgs/by-name/sw/sway-unwrapped/package.nix [hunk 3]
```diff
@@ -11,19 +11,16 @@
 , trayEnabled ? systemdSupport
 }:
 
-let
-  wlroots = wlroots_0_16;
```
Comment (norpol) [inline pkgs/by-name/sw/sway-unwrapped/package.nix:15]: might make sense to already pin it to the current wlroots version as sway will probably have to release a new version to follow upstream wlroots changes?

Comment (wineee) [reply pkgs/by-name/sw/sway-unwrapped/package.nix:15]: Should pin to `wlroots_0_17`, there will always be breaking changes before wlroots 1.0 is released

https://github.com/NixOS/nixpkgs/pull/245704#pullrequestreview-1553286765

**Review by wineee — CHANGES_REQUESTED** [review]

File: pkgs/by-name/sw/sway-unwrapped/package.nix (hunk 3 above)
Comment (raboof) [reply pkgs/by-name/sw/sway-unwrapped/package.nix:15]: probably makes sense, indeed the convention seems to be to update wlroots with a next version of sway

**Comment by norpol** [comment]
Thanks for bumping, just built it and verified with this that I'm running the new release

```
swaymsg -t get_version
sway version 1.9
```

for my configuration everything appears to be working as expected, screen lock works, xdg-desktop-portal-wlr/screen sharing works as well.

**Comment by trofi** [comment]
Result of `nixpkgs-review pr 287805` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>swayfx</li>
    <li>swayfx-unwrapped</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>nwg-panel</li>
    <li>nwg-panel.dist</li>
    <li>sway</li>
    <li>sway-contrib.grimshot</li>
    <li>sway-contrib.grimshot.man</li>
    <li>sway-unwrapped</li>
    <li>waybar</li>
    <li>wlprop</li>
  </ul>
</details>

**Review by eclairevoyant — COMMENTED** [review]
The change looks mostly good, but I believe we should also accordingly update [the module](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/programs/wayland/sway.nix#L121-L125) to use `wmenu` instead of `dmenu`

@trofi regarding the build failure, it's because swayfx is a fork based off the older sway, and hence only support wlr 0.16.x. Swayfx upstream needs to bump.

**Comment by wineee** [comment]
> Result of `nixpkgs-review pr 287805` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 2 packages failed to build:
> * swayfx
> * swayfx-unwrapped
> 
> 8 packages built:

swayfx still need wlroots_0_16, it shouldn't inherit sway's buildInput if it cannot be guaranteed to be released synchronously with sway

https://github.com/NixOS/nixpkgs/blob/cff84a2086451a31a552a672da95f132bd7c10af/pkgs/by-name/sw/swayfx-unwrapped/package.nix#L14

**Comment by eclairevoyant** [comment]
Created raboof/nixpkgs#2 to address the swayfx issues

**Comment by thiagokokada** [comment]
Any concerns about merging this PR and fixing `swayfx` in a subsequent PR?

**Comment by oxalica** [comment]
I'm running with sway on this PR for some days and it works fine. The only concern to me is that we are accepting parameter `wlroots_0_17`, which causes downstream overriding to be confusingly `.override { wlroots_0_17 = pkgs.wl_roots_0_16; /* or whatever */ }`. But I don't think it's fixable for by-name packages currently. Merging as-is is fine to me.

**Review by fpletz — APPROVED** [review]
I agree with the point https://github.com/raboof/nixpkgs/pull/2 that swayfx should be decoupled from sway eventually. Ff that PR is merged into this PR's branch, I'd say we can merge this for now and deal with swayfx later.

**Comment by raboof** [comment]
> But I don't think it's fixable for by-name packages currently.

I think I found a way - applied - if that seems to work let's merge like that, unless maintainers @primeos or @Synthetica9 have any input

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -4776,6 +4776,10 @@ with pkgs;
 
   slurp = callPackage ../tools/wayland/slurp { };
 
+  sway-unwrapped = callPackage ../by-name/sw/sway-unwrapped/package.nix {
```
Comment (thiagokokada) [inline pkgs/top-level/all-packages.nix:4779]: Not sure if mixing the old approach and `pkgs/by-name` is a good idea.

Comment (raboof) [reply pkgs/top-level/all-packages.nix:4779]: there are already 28 by-name packages where some parameters are overridden in all-packages.nix, so there seems to be precedent? do you know of any particular policy/problem discouraging it?

Comment (alyssais) [reply pkgs/top-level/all-packages.nix:4779]: The policy is actually to do it. https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#changing-implicit-attribute-defaults

**Comment by clushie** [comment]
> [...] [the module](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/programs/wayland/sway.nix#L121-L125) to use `wmenu` instead of `dmenu`

The file `nixos/modules/programs/wayland/sway.nix`: 

https://github.com/NixOS/nixpkgs/blob/ebe6e807793e7c9cc59cf81225fdee1a03413811/nixos/modules/programs/wayland/sway.nix#L122

needs to be changed to `wmenu` otherwise this PR will break the default configuration. See the diff:

```patch
diff /nix/store/v1dwgcx2mp0phzh1inymxgapizbi7psc-sway-1.9/etc/sway/config /nix/store/jik0q6a6v7q5l6k04r6f0h7zjig5a7vf-sway-1.8.1/etc/sway/config
21c21
< set $menu dmenu_path | wmenu | xargs swaymsg exec --
---
> set $menu dmenu_path | dmenu | xargs swaymsg exec --
208c208
<     status_command while date +'%Y-%m-%d %X'; do sleep 1; done
---
>     status_command while date +'%Y-%m-%d %I:%M:%S %p'; do sleep 1; done
```

On that front please also merge https://github.com/NixOS/nixpkgs/pull/284343 which was released shortly before `sway 1.9`

**Comment by raboof** [comment]
> needs to be changed to `wmenu`

updated, thanks!

> please also merge #284343

it looks like that was already merged last month? did you mean to refer to another PR?

**Comment by clushie** [comment]
@raboof Sorry I just double checked the commit and it should basically be `wmenu` and `dmenu` now 🙈, dmenu is still used to generate a list of executables in `$PATH` 

```
readlink -f `which dmenu_path`
/nix/store/hb7qbfbfdz5xw12fd9b37vyhkrcl6yd0-dmenu-5.2/bin/dmenu_path
```

**Review by eclairevoyant — APPROVED** [review]

**Pull request merged**